doc/compiler/README.md

```diff
@@ -0,0 +1,211 @@
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+## Table of Contents
+---
+
+When a method is chosen to be compiled, it is handed to the IL(Intermediate Language) generator first.
+The IL generator translates Java bytecodes to IL trees.
+
+Next, IL trees are processed by many independent optimization passes in the Optimizer.
+The optimizations first analyze the IL trees to find opportunities to optimize. Secondly,
+they transform the IL trees for better performance. Each method is compiled at different optimization
+levels based on its hotness such as cold, warm, etc.
+
+Code generators translate IL trees to assembly instructions for the target architecture.
+They first generate machine instructions wherein the result of each (value-producing) IL node is placed
+in a virtual register. Register Assignment (RA) replaces virtual registers with real registers.
+In the end code generators perform binary encoding to write the appropriate bits to the code cache buffer.
+
+```
+                                     Java Bytecode
+                                           |
+                    +----------------------+--------------------+
+                    | Compilation Thread   |                    |
+                    |           +----------v---------+          |
+                    |           |          IL        |          |
+ +-------------+    |           | +----------------+ |          |
+ | Compilation |    |           | |  IL Generator  | |          |
+ |   Control   +---->           | +----------------+ |          |
+ +------^------+    |           +----------+---------+          |
+        |           |                      |                    |
+        |           |     +----------------v----------------+   |
+ +------v-----+     |     |            Optimizer            |   |
+ |            |     |     | +----------+ +----------------+ |   |
+ |  Profiler  <----->     | | Analyses | |  Optimizations | |   |
+ |            |     |     | +----------+ +----------------+ |   |
+ +------------+     |     +----------------+----------------+   |
+                    |                      |                    |
+                    |  +-------------------v------------------+ |
+                    |  |            Code Generators           | |
+                    |  | +-----+ +---+ +---------+ +--------+ | |
+                    |  | | X86 | | Z | | PowerPC | | Aarch  | | |
+                    |  | +-----+ +---+ +---------+ +--------+ | |
+                    |  +-------------------+------------------+ |
+                    |                      |                    |
+                    +----------------------+--------------------+
++-----------------------------+      +-----+-----+
```

Comment:
Is there a connecting line missing from the JIT to the Metadata+Code ?

Reply:
Addressed in e8a650d